workspaces/icecap-host/Cargo.toml

```diff
@@ -30,3 +29,8 @@ resolver = "2"
 codegen-units = 1
 lto = true
 opt-level = 3
+
+[profile.dev]
+codegen-units = 1
+lto = true
```

Comment:
There are more differences between `dev` and `release` than `opt-level` and `codegen-units`. The defaults are (source: https://doc.rust-lang.org/cargo/reference/profiles.html):

```
[profile.dev]
opt-level = 0
debug = true
split-debuginfo = '...'  # Platform-specific.
debug-assertions = true
overflow-checks = true
lto = false
panic = 'unwind'
incremental = true
codegen-units = 256
rpath = false
```

```
[profile.release]
opt-level = 3
debug = false
split-debuginfo = '...'  # Platform-specific.
debug-assertions = false
overflow-checks = false
lto = false
panic = 'unwind'
incremental = false
codegen-units = 16
rpath = false
```

Is setting the `lto` to `true` needed to fix the issue?

Reply:
Well, these changes went through CI fine. So whatever needed to be changed was changed. Since this icecap issue has been longstanding, I don't think this it the PR to root-cause it.

Reply:
That's because CI calls `make -C /work/veracruz/workspaces icecap-tests PROFILE=release`
(https://github.com/veracruz-project/veracruz/blob/main/.github/workflows/main.yml#L61)

Reply:
I'm reverting this, as this is turning into too much trouble and this PR is not the place to figure this out